go/ct/gen/code_test.go

```diff
@@ -599,3 +601,54 @@ func TestVarCodeConstraintSolver_fitsOnUsed(t *testing.T) {
 		}
 	}
 }
+
+func TestCodeGenerator_IsDataConstraintSmallSize(t *testing.T) {
+	variable := Variable("X")
+
+	tests := []int{0, 1, 2}
+
+	for size := range tests {
+		t.Run(fmt.Sprintf("size: %v", size), func(t *testing.T) {
+
+			generator := NewCodeGenerator()
+			generator.AddIsData(variable)
+			generator.codeSize = new(int)
+			*generator.codeSize = size
+
+			assignment := Assignment{}
+
+			state, err := generator.Generate(assignment, rand.New(0))
+			if size < 2 {
+				if !strings.Contains(err.Error(), ErrUnsatisfiable.Error()) {
```

Comment:
nit
```suggestion
				if !errors.Is(err, ErrUnsatisfiable) {
```

Reply:
error from code generator in case of test contains more info than just error. https://github.com/Fantom-foundation/Tosca/blob/main/go/ct/gen/code.go#L138

Reply:
The line you are referencing is annotating an `ErrUnsatisifiable` with extra information, but it is still identifiable as such. Please study the [errors](https://pkg.go.dev/errors) package and the `%w` verb of [fmt.Errorf](https://pkg.go.dev/fmt#Errorf).